sonar-plugin-api/src/main/java/org/sonar/api/utils/Duration.java

```diff
@@ -121,7 +121,7 @@ public String encode(int hoursInDay) {
       stringBuilder.append(minutes);
       stringBuilder.append(MINUTE);
     }
-    return stringBuilder.toString();
+    return stringBuilder.length() == 0 ? "0" + MINUTE : stringBuilder.toString();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS864)